pages/dkp/kommander/1.3/projects/project-network-policies/index.md

```diff
@@ -0,0 +1,116 @@
+---
+layout: layout.pug
+beta: false
+navigationTitle: Project Network Policies
+title: Project Network Policies
+menuWeight: 8
+excerpt: Projects are created with a secure-by-default network policy and users needing more flexibility can edit or add more policies to tailor to their unique security needs.
+---
+
+Cluster networking is a critical and central part of Kubernetes that can also be quite challenging. All network communication within and between clusters depends on the presence of a Container Network Interface (CNI) plugin.
+
+# What are Network Plugins?
+Network plugins ensure that Kubernetes networking requirements are met and surface features needed by network administrators, such as enforcing network policies. Common Network Plugins include Flannel, Calico, and Weave among many others. As an example, D2iQ&reg; Konvoy&reg; uses the Calico CNI plugin by default, but can support others.
+
+Since pods are short-lived, the cluster needs a way to configure the network dynamically as pods are created and destroyed. Plugins provision and manage IP addresses to interfaces and let administrators manage IPs and their assignments to containers, as well as connections to more than one host, when needed.
+
+# What are Network Policies
+By default, and to enable fluid communications within and between clusters, all traffic is authorized between nodes and pods. Most production environments require some kind of traffic flow control at the IP address or port level. An application-centric approach to this uses Network Policies. Starting out pods are not isolated. They become isolated by having a Network Policy that selects them, and the policy limits the kind of traffic they can receive or send, depending on how it is configured. Network policies do not conflict because they are additive. Pods selected by more than one policy are subject to the union of the policies' ingress and egress rules.
+
+A Network Policy's rules define ingress and egress for network communications between pods and across namespaces. Successful traffic control using network policies is bi-directional. You have to configure both the egress policy on the source pod and the ingress policy on the destination pod to enable the traffic. If either end denies the traffic, it will not flow between the pods.
+
+Since the Kubernetes default is to allow all traffic, it's a common practice to create a default "deny all traffic" rule, and then specifically open up the pods, ports, and/or applications as needed.
+
+When you first create a Project, Kommander automatically creates a NetworkPolicy that allows traffic originating only within your Project namespace. Pods cannot talk to Pods outside of that namespace, by default. If you want to enable incoming traffic from other namespaces and pods, then you have to create a NetworkPolicy with ingress rules.
```

Comment:
I wonder if it's worth adding something about how any network policies that are created in a project are federated to all the project's target clusters in the project namespace? Or maybe overkill

Reply:
i don't think it's overkill at all! SamB would like us to limit use of the word "federated" in docs. Is there an alternative?